src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -339,60 +345,93 @@ ISymbol memberSymbol
 			);
 		}
 
+		private enum AssignmentQueryKind {
+			NothingToCheck,
+			ImmutabilityQuery,
+			Hopeless
+		}

```

Comment:
What I really wanted was something like:

```cs
private enum AssignmentQuery {
  NothingToCheck,
  Hopeless(Diagnostic),
  SubQuery(ImmutabilityQuery)
}
```

[which some languages support](https://doc.rust-lang.org/rust-by-example/custom_types/enum.html) but not C#.